### PR TITLE
[EMB-138] Developer handbook

### PR DESCRIPTION
**Pull request by aaxelb**
## Purpose
In preparation to welcome more developers into ember-osf-web, create a space within the repository for us to easily share knowledge, demo components, and read auto-generated docs.

And along the way, learn all about engines!

## Summary of Changes
Add a `handbook` in-repo engine that uses a fork of `ember-cli-addon-docs`. It's mostly empty, but has auto-generated docs and a page structure that seems like it should be useful. Some highlights:

### How to make the handbook more handy
screen shot 2018-05-24 at 14 14 20

### Component gallery
screen shot 2018-05-24 at 14 04 49

### 

### Fully indexed and ready to search
screen shot 2018-05-24 at 13 57 44

## Side Effects / Testing Notes
The handbook is disabled by default. To enable it locally:
1. add `HANDBOOK_ENABLED: true` to `ember-osf-web/config/local.js`
1. add `/handbook/` to the `routes` list in `osf.io/website/ember_osf_web/views.py`
1. go to `localhost:5000/handbook/`

### `{{link-to}}` customizations
ember-engines makes some changes to `link-to`'s behavior, so links within an engine are automatically namespaced. Unfortunately, that's clobbered by our own `link-to` customization (adding `ariaLabel` and `clickAction` attributes). Turns out our changes weren't actually necessary, though:
- `ariaLabel='Aria label!'` can be replaced by `(html-attributes aria-label='Aria label!')`
- `clickAction=(action 'clicked')` can be replaced by `click=(action 'clicked')`

So I removed our custom `link-to` and updated all uses of `ariaLabel` and `clickAction` accordingly.

## Ticket

https://openscience.atlassian.net/browse/EMB-138

# Reviewer Checklist

- [ ] meets requirements
- [ ] easy to understand
- [ ] DRY
- [ ] testable and includes test(s)
- [ ] changes described in `CHANGELOG.md`

### CODE REVIEW

**Comment by aaxelb** [comment]
@brianjgeiger Weird, they [pass on my travis](https://travis-ci.org/aaxelb/ember-osf-web/builds/383379335). But that is an error I get sometimes when rebuilding it locally, so I should be able to reproduce. Looking...

File: .eslintignore [hunk 1]
```diff
@@ -1,5 +1,8 @@
-coverage
-dist
-docs
-tmp
-blueprints/*/files
+!.eslintrc.js
```
Comment (jamescdavis) [inline .eslintignore]: :heart:

File: .eslintignore [hunk 2]
```diff
@@ -1,5 +1,8 @@
...
+!.template-lintrc.js
+/coverage
+/dist
+/docs
+/tmp
+/vendor
+/blueprints/*/files
```
Comment (jamescdavis) [inline .eslintignore]: Good call on anchoring these to the beginning of the path. These directory patterns should technically also have either a trailing slash or /**.

Comment (jamescdavis) [inline .eslintignore]: FYI: eslint itself uses /** in its own repo: https://github.com/eslint/eslint/blob/master/.eslintignore

**Pull request merged**